servlet.py

```diff
@@ -121,6 +124,49 @@ class BaseHandler(tornado.web.RequestHandler):
     def initialize(self):
         self.callback = self.get_argument('callback', default=None)
 
+    @classmethod
+    def initPairsGraph(cls):
+        for pair in cls.pairs:
+            lang1, lang2 = pair.split('-')
+            if lang1 in cls.pairs_graph:
+                cls.pairs_graph[lang1].append(lang2)
+            else:
+                cls.pairs_graph[lang1] = [lang2]
+
+    @classmethod
+    def calculatePaths(cls, start):
+        nodes = set()
+        for pair in map(lambda x: x.split('-'), cls.pairs):
+            nodes.add(pair[0])
+            nodes.add(pair[1])
+        dists = {}
+        prevs = {}
+        dists[start] = 0
+
+        while nodes:
+            u = min(nodes, key=lambda u: dists.get(u, float('inf')))
+            nodes.remove(u)
+            for v in cls.pairs_graph.get(u, []):
+                if v in nodes:
+                    other = dists.get(u, float('inf')) + 1        # TODO: weight(u, v)
```

Comment:
can you make it clear here whether high weight will lead to this edge being picked or low weight? (in the comment)